app/services/share_progress_variant_builder.rb

```diff
@@ -1,17 +1,17 @@
 require 'share_progress'
 
 class ShareProgressVariantBuilder
-  def self.create(params, variant_type:, page:, url:)
-    new(params, variant_type, page, url).create
+  def self.create(params:, variant_type:, page:, url:)
+    new(params, variant_type, page, url, nil).create
```

Comment:
on create it needs a url but won't have an id yet. on update it should already have an id but can't update the url cause it's just changing the variant, not the button.